pkg/kubelet/nodelease/controller_test.go

```diff
@@ -195,3 +198,207 @@ func TestNewLease(t *testing.T) {
 		})
 	}
 }
+
+type pair struct {
+	first, second interface{}
+}
+
+type fakeLeaseClient struct {
+	createCalls           chan pair
+	updateCalls           chan pair
+	deleteCalls           chan error
+	deleteCollectionCalls chan error
+	getCalls              chan pair
+	listCalls             chan pair
+	watchCalls            chan pair
+	patchCalls            chan pair
+}
+
+func newFakeLeaseClient() *fakeLeaseClient {
```

Comment:
You don't want to do that.
You want to use this one:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/kubernetes/typed/coordination/v1/fake/fake_coordination_client.go
instead.

Then you register Reactor:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/testing/fake.go#L102

and I think you don't need much more...

Reply:
I simplified the test. Reactor works perfectly.